src/Serialization/Fields.jl

```diff
@@ -48,3 +48,132 @@ function load_internal(s::DeserializerState, ::Type{gfp_fmpz_elem}, dict::Dict)
+# FqNmodfinitefield
+function save_internal(s::SerializerState, K::FqNmodFiniteField)
+    return Dict(
+        :def_pol => save_type_dispatch(s, defining_polynomial(K))
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{FqNmodFiniteField}, dict::Dict)
+    def_pol = load_type_dispatch(s, dict[:def_pol], check_namespace=false)
+    K, _ = FiniteField(def_pol, cached=false)
+
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{nf_elem, fq_nmod, Hecke.NfRelElem})
+    K = parent(k)
+    polynomial = parent(defining_polynomial(K))(k)
+    K_dict = save_type_dispatch(s, K)
+
+    return Dict(
+        :parent => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{nf_elem, fq_nmod, Hecke.NfRelElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent], check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial], check_namespace=false)
+    return K(polynomial)
+end
+
+################################################################################
+# Non Simple Extension
+function save_internal(s::SerializerState, K::Union{NfAbsNS, NfRelNS})
+    def_pols = defining_polynomials(K)
+    return Dict(
+        :def_pols => save_type_dispatch(s, def_pols),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNS, NfRelNS}},
+                       dict::Dict)
+    def_pols = load_type_dispatch(s, dict[:def_pols], check_namespace=false)
+    K, _ = NumberField(def_pols, cached=false)
+    return K
+end
+
+#elements
+function save_internal(s::SerializerState, k::Union{NfAbsNSElem, Hecke.NfRelNSElem})
+    K = parent(k)
+    polynomial = Oscar.Hecke.data(k)
+    polynomial_parent = parent(polynomial)
+    K_dict = save_type_dispatch(s, K)
+    return Dict(
+        :parent_field => K_dict,
+        :polynomial => save_type_dispatch(s, polynomial),
+        :polynomial_parent => save_type_dispatch(s, polynomial_parent)
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: Union{NfAbsNSElem, Hecke.NfRelNSElem}},
+                       dict::Dict)
+    K = load_type_dispatch(s, dict[:parent_field]; check_namespace=false)
+    polynomial = load_type_dispatch(s, dict[:polynomial]; check_namespace=false)
+    polynomial = evaluate(polynomial, gens(K))
+
+    return K(polynomial)
+end
+
+
+################################################################################
+# FracField
+function save_internal(s::SerializerState, K::FracField)
+    return Dict(
+        :base_ring => save_type_dispatch(s, base_ring(K)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       ::Type{<: FracField},
+                       dict::Dict)
+    R, _ = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
```

Comment:
Again (you may sense a theme here ;-)) I think this could be made "explicit", like this (bug again: untested):
```suggestion
                       ::Type{FracField{T}},
                       dict::Dict) where T
    R, _ = load_type_dispatch(s, T, dict[:base_ring])
```